app/components/polaris/base_checkbox.rb

```diff
@@ -31,6 +31,7 @@ def system_arguments
           @system_arguments[:aria][:checked] = "mixed"
         end
         opts[:class] = opts.delete(:classes)
+        @system_arguments[:form] = @form if @form.present? && !@attribute.present?
```

Comment:
It's better to use `opts` as we're within `tap` block.

```suggestion
        opts[:form] = @form if @form.present? && @attribute.blank?
```